### ISSUE TITLE
Is this also compatible with 1-2-4-6 Junkers heaters?

### ISSUE THREAD

**Issue by ltctceplrm** [Strooom/SmartTherm#11]
Like the TOP 30/375-2 ZBS

**Comment by Strooom** [comment]
Mmmm, I don't know this heater.. as it's 4 wires, it looks like it's using a different protocol or interface.
We need to know more info about what is on these 1-2-4-6 wires.

**Comment by ltctceplrm** [comment]
1 is for power
2 is a CAN Bus High
4 is ground
6 is CAN Bus Low
The thermostat is a TA270 with this wiring
![QPWlgL7](https://github.com/Strooom/SmartTherm/assets/14954927/b37a2263-72b3-4097-84ea-46cda65429b8)
![RUmD7FR](https://github.com/Strooom/SmartTherm/assets/14954927/0bc9a4f3-31af-4c7e-aa19-6a4e0154340c)
And this is what the wiring in the heater looks like:
![M3vKHou](https://github.com/Strooom/SmartTherm/assets/14954927/67553339-7b5a-44ff-a73d-0e5557649f62)

**Comment by nagyrobi** [comment]
CAN Bus is a completely different type of animal...

**Comment by Strooom** [comment]
Indeed, my design is intended for the older 'analog' thermostat interface. I have never used an ESP32 with CAN, but it seems the MCU has this interface.
So you could modify the design (hardware ánd software) to make it work for a heater with CAN.
Biggest challenge will be to find out what messages/protocol to talk to the heater.

**Comment by nagyrobi** [comment]
Interesting that Junkers/Bosch used EMS protocol (proprietary interface) so far, since when they switched to CAN?

**Comment by nagyrobi** [comment]
But, given the fact that the heater has the connector labelled as **1-2-4**-6, there are chances that you can use the analog solution provided by this hardware on the 1-2-4 pins only. You have to check in the service/installer manual of your heater, if the communication protocol of the heater can be switched from CAN 1-2-4-6 to 0-24V 1-2-4.
In my case (a Junkers Ceraclass) it was possible to choose between EMS and 0-24V 1-2-4. Of course the original thermostat won't work in this case.

**Comment by ltctceplrm** [comment]
>  So you could modify the design (hardware ánd software) to make it work for a heater with CAN.
Biggest challenge will be to find out what messages/protocol to talk to the heater.

I think that would be well above my skill level unfortunately, especially because I don't know how long my heater will still last. Next one I'll definitely find one that is compatible with every smart system possible 😉 

> Interesting that Junkers/Bosch used EMS protocol (proprietary interface) so far, since when they switched to CAN?

My heater (Junkers TOP 30/375-2 ZBS) uses Heatronic 2 and I think EMS is Heatronic 3-4 right? I think my model might be too old then perhaps.



> But, given the fact that the heater has the connector labelled as **1-2-4**-6, there are chances that you can use the analog solution provided by this hardware on the 1-2-4 pins only. You have to check in the service/installer manual of your heater, if the communication protocol of the heater can be switched from CAN 1-2-4-6 to 0-24V 1-2-4. In my case (a Junkers Ceraclass) it was possible to choose between EMS and 0-24V 1-2-4. Of course the original thermostat won't work in this case.

That would be very interesting but will it still support modulation?

**Comment by nagyrobi** [comment]
The modulation is done by this circuit!

**Comment by ltctceplrm** [comment]
I looked at the manual for the heater and not the thermostat and they do mention that there are two ways of connecting a thermostat, one of them is for the TR-100 with the 1-2-4 cables and the other is with for the TA-270 with the cables like I have.
Here's an image from the manual
![image](https://github.com/Strooom/SmartTherm/assets/14954927/262e9ffa-2409-47d7-8a2f-57a0c9134cbe)

If I understand correctly I could just unplug the cables going to the green section on the right and instead connect the smarttherm board to the white section on the left, with cables 1-2-4:
![image](https://github.com/Strooom/SmartTherm/assets/14954927/8d10348e-b5af-4c07-8005-ec4b6f27463e)

**Comment by Strooom** [comment]
Yes, the manual also mentions the TR21 and this was the thermostat that I reverse engineered and for which the hardware is providing compatible signals.
